soda_machine.rb

```diff
@@ -7,12 +9,31 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.length # or .size
   end
 
   def find_soda(soda_brand)
+    @sodas.find {|soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
+    return nil unless
+      @sold_soda = find_soda(soda_brand)
+      @sodas.delete(@sold_soda)
+      @cash += @sold_soda.price
+      @sold_soda
+    end
+
```

Comment:
Since the method ```delete``` deletes the soda and returns it, you can have it as the last statement as shown below. 

```ruby
 @sold_soda = find_soda(soda_brand)
 @cash += @sold_soda.price
 @sodas.delete(@sold_soda)
```